### ISSUE TITLE
Data structures

### ISSUE THREAD

**Issue by adriatic** [aurelia-ui-toolkits/bridge-catalog-admin#4]
### Component

Any element from this collection

![image](https://cloud.githubusercontent.com/assets/2712405/22434887/f0c52672-e6eb-11e6-9672-146b00dd93cc.png)

It is defined as
```
namespace Catalog.Models
{
    public class Component
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime DateCreated { get; set; }
        public string  UserName { get; set; }
        public ICollection<Sample> Samples { get; set; }
    }
}
```

### Sample

![image](https://cloud.githubusercontent.com/assets/2712405/22435113/bdd4113c-e6ec-11e6-8b57-5609faa019da.png)

**Comment by Thanood** [comment]
I think it may be beneficial to store the creation data (date, user) with each sample instead of the component. So imo a component consists of an id, name and a collection of samples while the samples have properties for an id, name creation date, user and a gist reference.

**Comment by adriatic** [comment]
Yes, @Thanood - I agree with you and your proposed "data set" seems fine to start with. The time-stamps and user IDs could be correlated with GitHub data and the tracing in the case of a problem would be easier
